packages/ds-ext/src/duckDBStorage.ts

```diff
@@ -0,0 +1,97 @@
+import { ObserverStorage } from '@data-story/core';
+import { Database } from 'duckdb-async';
+
+export class DuckDBStorage implements ObserverStorage {
+  private db: Database | null = null;
+  private dbPath: string;
+
+  constructor(dbPath: string) {
+    this.dbPath = dbPath;
+    this.initDatabase();
+  }
+
+  async initDatabase() {
+    this.db = await Database.create(this.dbPath);
+    await this.db.all(`
+      CREATE TABLE IF NOT EXISTS linkCounts (
+        linkId TEXT PRIMARY KEY,
+        count INT,
+        createTime TIMESTAMP,
+        updateTime TIMESTAMP
+      );
+    
+      CREATE TABLE IF NOT EXISTS linkItems (
+        linkId TEXT,
+        item JSON,
+        createTime TIMESTAMP,
+        updateTime TIMESTAMP
+      );
+      CREATE INDEX idx_linkId ON linkItems (linkId);
+    
+      CREATE TABLE IF NOT EXISTS nodes (
+        nodeId TEXT PRIMARY KEY,
+        status TEXT,
+        createTime TIMESTAMP,
+        updateTime TIMESTAMP
```

Comment:
Do we have a use case for updateTime or is it just for a uniform/standard setup to have it?

Reply:
The `updateTime` field is primarily used for a uniform/standard setup. 
However, it can also serve as a data observation tool during development. For instance, in the `node` table, we can track the time taken for a node to progress from the initial state to available status and then to complete status using `createTime` and `updateTime`.